include/envoy/server/configuration.h

```diff
@@ -65,6 +65,27 @@ class Watchdog {
   actions() const PURE;
 };
 
+class StatsConfig {
+public:
+  virtual ~StatsConfig() = default;
+
+  /**
+   * @return std::list<Stats::SinkPtr>& the list of stats sinks initialized from the configuration.
+   */
+  virtual std::list<Stats::SinkPtr>& sinks() PURE;
```

Comment:
Make this method return a const reference to the list, and also a const method in general. Then, you can populate the list directly into the StatsConfigImpl via an impl specific method.